tests/libvmifact/factory.go

```diff
@@ -146,3 +148,23 @@ func NewWindows(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 	vmi.Spec.Domain.Firmware = &kvirtv1.Firmware{UUID: WindowsFirmware}
 	return vmi
 }
+
+func NewTinyOSVMWithDataVolume(dv *cdiv1.DataVolume, extraVMOpts ...libvmi.VMOption) *kvirtv1.VirtualMachine {
+	// Supplied with no user data, AlpimeWithTestTooling image takes more than 200s to allow login
+	withNonEmptyUserData := libvmi.WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")
+	opts := []libvmi.VMOption{
+		libvmi.WithDataVolumeTemplate(dv),
```

Comment:
You will pass this as input anyway, so passing it as an input option will be the same "cost".

Reply:
Yeah cost will be the same, but the idea was to require as little knowledge as possible from a casual user of this trying to get a persistent disk VM

Reply:
A tester author needs to know how to build a VM/VMI manifest, that is the requirement.
Asking him to know that in order to add a datavolume template, he needs to call a specific builder is reasonable IMO.